.github/workflows/python_wheels.yml

```diff
@@ -173,11 +182,20 @@ jobs:
         run: |
           os_image=${{ matrix.os_image }}
           distr=`echo $os_image | cut -d : -f 1 | cut -d / -f 1`
-          os_version=`echo ${os_image#$distr} | cut -c2-`
+          if ${{ inputs.environment != '' }}; then
+            # Validate against the public images
+            base_image=$os_image
+          else
+            # If we are not staging, we instead validate the images
+            # pushed to GHCR to limit our pull rate on public registries.
+            repo_owner=${{ github.repository_owner }}
+            image_tag=`echo $os_image | sed -e 's@.*\.io/\(\)@\1@'`
+            base_image=ghcr.io/${repo_owner,,}/${image_tag#${repo_owner,,}}
+          fi
```

Comment:
Just thinking out loud ... it might be nice to have some sort of manual override to allow us to fall back on the public registries in case ours get deleted and/or out of date for some reason. If that happens, though, we could presumably just manually run the gh_registry workflow to repopulate, right? I'll leave that up to you.

Reply:
Yes, exactly, the idea is to manually run the gh_registry workflow. The override would require dragging a configuration through all edited workflows, so I don't want to do that pre-emptively. If we see we need it, I'll add it.